src/pages/Encounters/tabs/EncounterNotesTab.tsx

```diff
@@ -0,0 +1,373 @@
+
+  return (
+    <div
+      className={cn(
+        "flex w-full mb-4 animate-in fade-in-0 slide-in-from-bottom-4",
+        isCurrentUser ? "justify-end" : "justify-start",
+      )}
+    >
+      <div
+        className={cn(
+          "flex max-w-[80%] items-start gap-3",
+          isCurrentUser ? "flex-row-reverse" : "flex-row",
+        )}
+      >
+        <div className="flex-shrink-0 mt-1">
+          <UserAvatar user={message.created_by} />
+        </div>
+
+        <div
+          className={cn(
+            "flex flex-col",
+            isCurrentUser ? "items-end" : "items-start",
+          )}
+        >
+          <span className="text-xs text-muted-foreground mb-1">
+            {message.created_by.username}
+          </span>
+          <div
+            className={cn(
+              "p-3 rounded-lg break-words",
+              isCurrentUser
+                ? "bg-primary-100 text-primary-foreground ml-auto rounded-br-none"
+                : "bg-muted mr-auto rounded-bl-none",
+            )}
+          >
+            <p className="whitespace-pre-wrap">{message.message}</p>
+          </div>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+interface PaginatedResponse<T> {
+  count: number;
+  next: string | null;
+  previous: string | null;
+  results: T[];
+}
+
+export const EncounterNotesTab = ({ encounter }: EncounterTabProps) => {
+  const { t } = useTranslation();
+  const queryClient = useQueryClient();
+  const [selectedThread, setSelectedThread] = useState<string | null>(null);
+  const [showNewThreadForm, setShowNewThreadForm] = useState(false);
+  const [newThreadTitle, setNewThreadTitle] = useState("");
+  const [newMessage, setNewMessage] = useState("");
+  const LIMIT = 20;
+  const { ref, inView } = useInView();
+
+  // Fetch threads
+  const { data: threadsData, isLoading: threadsLoading } = useQuery({
+    queryKey: ["threads", encounter.id],
+    queryFn: query(routes.notes.patient.listThreads, {
+      pathParams: { patientId: encounter.patient.id },
+      queryParams: { encounter: encounter.id },
+    }),
+  });
+
+  // Auto-select the first thread when data is loaded
+  useEffect(() => {
+    if (threadsData?.results.length && !selectedThread) {
+      setSelectedThread(threadsData.results[0].id);
+    }
+  }, [threadsData, selectedThread]);
+
+  // Fetch messages with offset pagination
+  const {
+    data: messagesData,
+    isLoading: messagesLoading,
+    hasNextPage,
+    fetchNextPage,
+    isFetchingNextPage,
+  } = useInfiniteQuery<PaginatedResponse<Message>>({
+    queryKey: ["messages", selectedThread],
+    queryFn: async ({ pageParam = 0 }) => {
+      const response = await query(routes.notes.patient.getMessages, {
+        pathParams: {
+          patientId: encounter.patient.id,
+          threadId: selectedThread!,
+        },
+        queryParams: {
+          limit: String(LIMIT),
+          offset: String(pageParam),
+        },
+      })({ signal: new AbortController().signal });
```

Comment:
won't we get `signal` from the `queryFn`'s callbacks args itself? 🤔 

```suggestion
    queryFn: async ({ pageParam = 0, signal }) => {
      const response = await query(routes.notes.patient.getMessages, {
        pathParams: {
          patientId: encounter.patient.id,
          threadId: selectedThread!,
        },
        queryParams: {
          limit: String(LIMIT),
          offset: String(pageParam),
        },
      })({ signal });
```